### ISSUE TITLE
BUG: If you provide a row with a property named 'value', it breaks.

### ISSUE THREAD

**Issue by QuentinFchx** [swimlane/ngx-datatable#993]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, post on Stackoverflow or Gitter
```

**Current behavior**

If you provide a row with a property `value`, it breaks.

**Expected behavior**

It shouldn't 😫 

**Reproduction of the problem**

http://plnkr.co/edit/7vlf5Xk4G085TM2OconV?p=preview

**Please tell us about your environment:**

* **Table version:** 0.8.x

10.2.3

* **Angular version:** 2.0.x

4.x

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

Relates to https://github.com/swimlane/ngx-datatable/issues/950

**Comment by QuentinFchx** [comment]
@amcdnl I think the issue comes from the `datatable-body` component:
Within the `updateRows` method, the `temp` array is populated with either "group of rows" or "rows". Each group or rows is of the form `{key, value}` meaning that we can have the following structure for the `temp` prop (considering we use `{value: any}` as a row definition):
- Groups of rows: `{key: any, value: {value: any}[]}[]`
- Rows: `{value: any}[]`
It then conflicts within the template where we have the following (simplified):
```
*ngFor="let group of temp"
    *ngIf="group.value"
    *ngFor="let row of group.value"   <-- in our case, this is a real value and not the value of the row = it breaks
```

This needs to be refactored. A quick-fix would be to rename the rowGroup definition (`{key: any, $$value: any[]}`) but I think we can come up with something more elegant

**Comment by QuentinFchx** [comment]
Here is a quickfix for anyone impacted:
```ts
import { DataTableBodyComponent } from '@swimlane/ngx-datatable';
const _updateRows = DataTableBodyComponent.prototype.updateRows;
DataTableBodyComponent.prototype.updateRows = function () {
    _updateRows.call(this);
    if (!this.groupedRows) this.temp = [{ key: -1, value: this.temp }];
};
```

**Comment by QuentinFchx** [comment]
This has been fixed through #1032
